### PR TITLE
Add `stylelint-selector-max-depth` plugin to the docs plugin list.

### PR DESCRIPTION
**Pull request by dryoma**
The discussion: #1141

### CODE REVIEW

**Comment by evilebottnawi** [comment]
@dryoma be good add travis CI https://travis-ci.org/

**Comment by dryoma** [comment]
@evilebottnawi done, am I good now? :)

**Comment by evilebottnawi** [comment]
@dryoma LGTM

File: docs/user-guide/plugins.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ Plugins are rules built by the community that support methodologies, toolsets, *
 ### Selector
 
 - [`stylelint-selector-bem-pattern`](https://github.com/davidtheclark/stylelint-selector-bem-pattern): Specify a BEM pattern for selectors (incorporates [postcss-bem-linter](https://github.com/postcss/postcss-bem-linter)).
+- [`stylelint-selector-max-depth`](https://github.com/dryoma/stylelint-selector-max-depth): Limit selectors depth.
```
Comment (davidtheclark) [inline docs/user-guide/plugins.md]: I think a more complete description would be helpful.

Comment (dryoma) [inline docs/user-guide/plugins.md]: Done. How about now?

File: docs/user-guide/plugins.md [hunk 2]
```diff
@@ -7,6 +7,7 @@ Plugins are rules built by the community that support methodologies, toolsets, *
 ### Selector
 
 - [`stylelint-selector-bem-pattern`](https://github.com/davidtheclark/stylelint-selector-bem-pattern): Specify a BEM pattern for selectors (incorporates [postcss-bem-linter](https://github.com/postcss/postcss-bem-linter)).
+- [`stylelint-selector-max-depth`](https://github.com/dryoma/stylelint-selector-max-depth): Limit selectors depth. In terms of this rule, selector depth is how many levels of HTML structure (not necesserily direct descendants) the selector is reflecting. For example, `.foo h3 + a > span` has the depth of 3: `.foo` is level 1, `a` (and `h3`, they are siblings) is level 2, and `span` is level 3.
```
Comment (davidtheclark) [inline docs/user-guide/plugins.md]: That looks good to me. Could add an apostrophe, though: `Limit selectors' depth.` @jeddy3 What do you think?

**Comment by jeddy3** [comment]
>  @jeddy3 What do you think?

You guys made some [good](https://github.com/stylelint/stylelint/issues/1141#issuecomment-214735531) [arguments](https://github.com/stylelint/stylelint/issues/1141#issuecomment-215062199) for this to be included in core. I'm convinced and happy to see this in core if @dryoma is willing to port it over?

**Comment by dryoma** [comment]
Sure, starting to work on the port.

**Pull request closed without merging**